CMakeLists.txt

```diff
@@ -139,6 +139,29 @@ endif()
 # Default to using the static CRT
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
+# Find the Git branch & tag info for use in Mock ICD
+set(GIT_BRANCH_NAME "--unknown--")
+set(GIT_TAG_INFO "--unknown--")
+find_package (Git)
+if (GIT_FOUND AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/.git/HEAD")
```

Comment:
Your CMake and C++ code don't match intent.

The CMake code will pass even if it doesn't find git or the .git folder.

Yet the C++ code requires GIT_BRANCH_NAME and GIT_TAG_INFO be defined.

IE make the C++ code match the permissive nature of the CMake code.

Reply:
The `GIT_BRANCH_NAME` and `GIT_TAG_INFO` variables in CMake are always defined, they will just default to `--unknown--`. This allows the lack of a .git folder to gracefully fallback to predefined values.

Now, you could argue that the `target_compile_definitions(VkICD_mock_icd PRIVATE GIT_BRANCH_NAME="${GIT_BRANCH_NAME}" GIT_TAG_INFO="${GIT_TAG_INFO}")` should be hidden behind an if, as well as the C++ code behind a `#if`.

Reply:
I was mainly trying to articulate that this will break the gn build. Due to the C++ code relying on the CMake code.

Reply:
Oh. You should have said that then. I read that as just 'an inconsistency of intent'.

Reply:
I have added a #if defined() #else #endif to handle the lack of definition in the code. Thanks for clearing up my misunderstanding.